docker/data-science/notebooks/ch10/2.judgments-and-logging.ipynb

```diff
@@ -150,40 +152,49 @@
    "metadata": {},
    "outputs": [
     {
-     "data": {
-      "text/plain": [
-       "{'responseHeader': {'status': 0, 'QTime': 3}}"
-      ]
-     },
-     "execution_count": 5,
-     "metadata": {},
-     "output_type": "execute_result"
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[\n",
+      "  {\n",
+      "    \"name\": \"title_bm25\",\n",
+      "    \"store\": \"movies\",\n",
+      "    \"class\": \"org.apache.solr.ltr.feature.SolrFeature\",\n",
+      "    \"params\": {\n",
+      "      \"q\": \"title:(${keywords})\"\n",
+      "    }\n",
+      "  },\n",
+      "  {\n",
+      "    \"name\": \"overview_bm25\",\n",
+      "    \"store\": \"movies\",\n",
+      "    \"class\": \"org.apache.solr.ltr.feature.SolrFeature\",\n",
+      "    \"params\": {\n",
+      "      \"q\": \"overview:(${keywords})\"\n",
+      "    }\n",
+      "  },\n",
+      "  {\n",
+      "    \"name\": \"release_year\",\n",
+      "    \"store\": \"movies\",\n",
+      "    \"class\": \"org.apache.solr.ltr.feature.SolrFeature\",\n",
+      "    \"params\": {\n",
+      "      \"q\": \"{!func}release_year\"\n",
+      "    }\n",
+      "  }\n",
+      "]\n"
+     ]
     }
    ],
    "source": [
     "engine.delete_feature_store(tmdb_collection, \"movies\")\n",
     "\n",
     "feature_set = [\n",
-    "    {\n",
-    "      \"name\": \"title_bm25\",\n",
-    "      \"store\": \"movies\",\n",
-    "      \"class\": \"org.apache.solr.ltr.feature.SolrFeature\",\n",
-    "      \"params\": {\"q\": \"title:(${keywords})\"}\n",
-    "    },\n",
-    "    {\n",
-    "      \"name\": \"overview_bm25\",\n",
-    "      \"store\": \"movies\",\n",
-    "      \"class\": \"org.apache.solr.ltr.feature.SolrFeature\",\n",
-    "      \"params\": {\"q\": \"overview:(${keywords})\"}\n",
-    "    },\n",
-    "    {\n",
-    "      \"name\" : \"release_year\",\n",
-    "      \"store\": \"movies\",\n",
-    "      \"class\": \"org.apache.solr.ltr.feature.SolrFeature\",\n",
-    "      \"params\": {\"q\": \"{!func}release_year\"}}]\n",
+    "    ltr.generate_query_feature(\"title_bm25\", \"movies\", \"title\"),\n",
+    "    ltr.generate_query_feature(\"overview_bm25\", \"movies\", \"overview\"),\n",
+    "    ltr.generate_field_value_feature(\"release_year\", \"movies\", \"release_year\")]\n",
     "\n",
-    "resp = engine.create_feature_store(tmdb_collection, feature_set)\n",
-    "resp.json()"
+    "engine.upload_features(tmdb_collection, feature_set)\n",
```

Comment:
if it's going to be `engine.upload...` instead of `ltr.upload...`, then I think we should add `ltr` to the function name.

i.e. `ltr.upload_model` + `ltr.upload_feature_set` works or `engine.upload_ltr_model` + `engine.upload_ltr_feature_set` works.

Many of the things we're doing are "models" being used (signals_boosting model, collaborative filtering model, and click model, user personalization model, etc.), so a generic `engine.upload_model` or `engine.upload_feature_set` feel weird to me because this is specific to LTR only. In the book, the models we're uploading are "linear ranking classifier" models, and everything else we use we either use directly in the notebooks or upload via other means (directly into collections).

Some search engines have generic support for "models", but they have to be marshalled into a specific format in those cases (like ONYX of Neural Networks or architecture-dependent models depending on the user case. In elastic, for instance: https://www.elastic.co/guide/en/machine-learning/current/ml-nlp-model-ref.html)

This is why I was suggesting we just have the model uploading in the "ltr" object versus the "engine" object, just to keep it super tightly focused on the only think we're doing here (LTR). having `engine.upload_ltr_model` and `engine.upload_ltr_feature_set` functionally does the same thing, though.